### ISSUE TITLE
about the compress texture mipmap.

### ISSUE THREAD

**Issue by RaoGY** [TimvanScherpenzeel/texture-compressor#15]
First, thank you very much! 
   The code is very useful for me. While I have found a problem about the mipmap. when the image is not a square, we have to generate the min level to 1X1. But I find that you check the level only by image width and ignore the importance of height, resulting in the min mipmap of  1X2, the appearance of warning in WebGL , and some invalid texture.

**Comment by TimvanScherpenzeel** [comment]
In the current implementation all passed in textures must be square and a power of two, did the program not throw any errors?

It should have thrown an error in: https://github.com/TimvanScherpenzeel/texture-compressor/blob/00c1714db0ed01850677a0b487028558c8de15a3/lib/index.js#L16-L21

It does throw the following for me:
```
node ./bin/texture-compressor -i ./docs/example/example-2.png -o ./docs/example/flippedY/example-astc-4x4-2.ktx -t astc -c astc -b 4x4 -y
/Users/TVS/Projects/texture-compressor/lib/index.js:20
	throw new Error('Input image must be square');
	^

Error: Input image must be square
    at Object.<anonymous> (/Users/TVS/Projects/texture-compressor/lib/index.js:20:8)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/TVS/Projects/texture-compressor/bin/texture-compressor.js:2:1)
    at Module._compile (module.js:652:30)
```

**Comment by RaoGY** [comment]
My image is a power of two but not a square. so it do not throw any error. I have modified the code and solve my problem. Thank you!
![image](https://user-images.githubusercontent.com/8901500/43560466-358eeb6c-9645-11e8-9379-cab5ad0a50a5.png)

**Comment by TimvanScherpenzeel** [comment]
My apologies, I see my mistake now! I'll do a bit more testing Thank you raising the issue and proposing a solution :+1:

**Comment by TimvanScherpenzeel** [comment]
Hi @MrRender, unfortunately due to the lack of automatic resizing in `texture-compressor` I cannot support textures that are not square and not a power of two as I cannot guarantee the current tools accept non-power of two textures. Maybe this will change in the future.

In my confusion I used `&&` instead of `||` when I initially wrote the tool.

**Comment by TimvanScherpenzeel** [comment]
Fixed and merged in #18 .
